content/js1/sprints/2/prep/index.md

```diff
@@ -8,4 +8,406 @@ backlog= 'Module-JS1'
+- Give examples of when the `undefined` data type may occur in a JavaScript program
+- Construct simple expressions to represent a set of operations
+- Define a _parameter_
+- Identify the return value a call site given its function declaration
+- Identify the value assigned to a parameter when a function is invoked
+- Identify the type of a return value given some call site
+- Given a short program, state the number of times a given function is called
+- Differentiate between parameters and arguments
+- Invoke a given function with appropriate arguments to produce some target output
+- Describe how to log the output of a function
+- Analyse each line in a given program and state what the line does when the program executes
+- Explain the difference between returning a value and logging a value
+- Assess and explain whether a function can reference a given variable or not
+- Explain what is meant by the term "implementation opacity"
+- Explain how to reuse some code to perform a specific task
+- Propose clear and meaningful parameter names for a function
+- Given a problem, suggest clear semantic names for a function designed to solve the given problem
+- Give examples of how functionality can be generalised
+
+## Notes
+
+### 🧩 Percentages
+
+Given a number I want to convert it into a percentage format.
+
+For example, given the number `0.5` we should get `"50%"`
+And given the number `0.23` we should get `"23%"` and so forth.
+
+### Restating the problem
+
+We want to implement some functionality to convert _any_ given number into a percentage.
+Previously, we saw how to use some functions to perform a given task. Here are some examples:
+
+```js {linenos=table,linenostart=1}
+console.log("hello world"); // prints "hello world" to the console
+Math.round(3.141); // will output 3
+```
+
+All these expressions are _function calls_: we’re passing input to the functions, `console.log` or `Math.round` to use their functionality. In the case of `Math.round` and `console.log` these functions have been created and made available to use whenever we start running a program with NodeJS.
+
+However, there is so no such pre-built function that allows us to convert any number to a percentage. So we're going to create a function called `convertToPercentage` with the following properties:
+
+- takes any number input
+- converts it to a string output in percentage form.
+
+```js {linenos=table,linenostart=1}
+convertToPercentage(0.5); // should return "50%"
+```
+
+```js {linenos=table,linenostart=1}
+convertToPercentage(0.23); // should return "23%"
+```
+
+### Useful expressions
+
+It is often helpful to solve a problem in one specific instance before doing it for all cases.
+
+In programming, we always try the simplest thing first. Let’s consider how to convert just one number in to a percentage.
+Let’s consider the variable declaration below:
+
+```js {linenos=table,linenostart=1}
+const num = 0.5;
```

Comment:
Throughout, I think we should use the variable name `decimal` or `decimalNumber` or similar here, rather than `num`. "num" could easily be a % just as a number without a % after (i.e. `const num = 100;` could be meant to become `"100%"`). Let's make an example of how we can make that more clear though naming :)